### PR TITLE
Feature/api field list

### PR DESCRIPTION
**Pull request by eth8505**

### CODE REVIEW

**Review by prox2k — APPROVED** [review]

File: src/classes/Api.php [hunk 1]
```diff
@@ -160,4 +167,29 @@ public function reset() {
             $this->adapter = NULL;
         }
 
+        /**
+         * @inheritdoc
+         */
+        public function getFormFields($recipientlistId)
+        {
+
+            $cacheFilePath = get_temp_dir() . 'rapidmail_fields.php';
```
Comment (dendemann) [inline src/classes/Api.php]: I think you'd need a unique filename (per installation) here as get_temp_dir() might fallback to sys_get_temp_dir() which could mess up caches on shared systems?!

File: src/classes/Api.php [hunk 2]
```diff
@@ -160,4 +167,29 @@ public function reset() {
...
+            if (!is_file($cacheFilePath) || (time() - filemtime($cacheFilePath)) > self::API_FIELDS_CACHE_MAXAGE_SECONDS) {
+
+                $cacheData = $this->adapter()->getFormFields($recipientlistId);
+
+                if (false === file_put_contents($cacheFilePath, '<?php return '. var_export($cacheData, true) . ';')) {
+                    throw new \RuntimeException('File "' . $cacheFilePath . '" could not be written');
+                }
+
+            } else {
+                $cacheData = require $cacheFilePath;
```
Comment (dendemann) [inline src/classes/Api.php]: Might be a security issue if someone else puts evil code into TEMP_DIR/rapidmail_fields.php`

File: src/classes/Api.php (hunk 1 above)
Comment (eth8505) [reply src/classes/Api.php]: Changed to JSON.

File: src/classes/Api.php (hunk 2 above)
Comment (eth8505) [reply src/classes/Api.php]: Changed to JSON.

**Review by dendemann — APPROVED** [review]

**Pull request merged**